presto-main/src/main/java/com/facebook/presto/execution/SqlQueryManager.java

```diff
@@ -343,6 +344,21 @@ public QueryInfo createQuery(Session session, String query)
         return queryInfo;
     }
 
+    public static Statement unwrapExecuteStatement(Statement statement, SqlParser sqlParser, Session session)
+    {
+        if (!(statement instanceof Execute)) {
+            return statement;
+        }
+
+        String sqlString = session.getPreparedStatementFromExecute((Execute) statement);
+        Statement unwrappedStatement = sqlParser.createStatement(sqlString);
+        if (unwrappedStatement instanceof Execute) {
+            throw new PrestoException(NOT_SUPPORTED, "EXECUTE of EXECUTE statements is not supported");
```

Comment:
Let's remove "statements" from this message: "EXECUTE of EXECUTE is not supported"